risks/MASVS-STORAGE/1-store-sensitive-data-securely/data-unencrypted-internal/android-data-in-sandbox/test.md

```diff
@@ -0,0 +1,30 @@
+---
+platform: android
+title: Sensitive Data Written to Private Data Directory (Sandbox) Unencrypted.
+type: [dynamic, filesystem]
+mitigations:
+- android-use-keystore
+- android-use-androidx-security
+prerequisites:
+- identify-sensitive-data
+---
+
+## Steps
+
+1. Start the device.
+
+2. Launch and use the app going through the various workflows while inputting sensitive data wherever you can. Taking note of the data you input can help identify it later using tools to search for it.
+
+3. Take a copy of the app's private data directory for offline analysis. Using tar will preserve the filesystem structure and permissions.
+
+4. Search the extracted data for items such as keys, passwords and any sensitive data inputted into the app.
+
+5. Check files for sensitive data that has been encoded with algorithms such as base64 which obscures but does not protect sensitive data.
+
+## Observation
+
+Files within the private data directory contain sensitive data.
+
+## Evaluation
+
+The test case fails if you find sensitive data in the app's private data directory which has not been encrypted with strong cryptography. This includes plaintext data as well as encoding such as base64 or obfuscation such as xoring.
```

Comment:
The [observation](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit#heading=h.h9gqgz4hdubj) should describe the resulting data that is going to be evaluated. For example, the "extracted data from the app's private data directory properly decoded, uncompressed, etc."

Then the evaluation could be what step 4 was. Something like:

> Search the extracted data for items such as keys, passwords and any sensitive data inputted into the app. The test case fails if you find any of this sensitive data.